File: Problems/Codeforces/MirrorGrid.py

def rotate(i, j):
    return j, N - i - 1

T = int(input())
for _ in range(T):

    N = int(input())

    A = []
    for i in range(N):
        A.append(input())

    ans = 0
    for i in range(N):
        for j in range(N):
            x = i; y = j; s = 0
            for _ in range(4):
                s += int(A[x][y])
                x, y = rotate(x, y)
            ans += min(s, 4 - s)

    print(ans // 4)

File: Problems/Codeforces/MirrorGrid.cpp
#include <bits/stdc++.h>
using namespace std;

int T, N;
vector<string> A;

pair<int, int> rotate(int i, int j)
{
    return {j, N - i - 1};
}

int main()
{
    cin >> T;
    while (T--)
    {
        cin >> N;

        A.resize(N);
        for (int i = 0; i < N; i++) cin >> A[i];
        
        int ans = 0;
        for (int i = 0; i < N; i++) for (int j = 0; j < N; j++)
        {
            int x = i, y = j, s = 0, times = 4;
            while (times--)
            {
                s += (A[x][y] - '0');
                tie(x, y) = rotate(x, y);
            }
            ans += min(s, 4 - s);
        }

        cout << ans / 4 << '\n';
    }
}
